### PR TITLE
Feature/new watchlist btn

### PR DESCRIPTION
**Pull request by devanshu-santiment**

### CODE REVIEW

**Comment by devanshu-santiment** [comment]
**Desktop View**
image
**Mobile View**
image

File: src/components/Watchlists/NewWatchlistForm.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    this.setState({ name: e.target.value })
+  }
+
+  togglePrivacy = () => {
+    this.setState({ isPublic: !this.state.isPublic })
+  }
+
+  onCreateWatchlist = () => {
+    const { name, isPublic } = this.state
+    this.props.createWatchlist({
```
Comment (Partysun) [inline src/components/Watchlists/NewWatchlistForm.js]: You can pass ...state

**Comment by Partysun** [comment]
But let's think about each cases.

What will happen if Error, when we try to create watchlist?
When we click on create, do we close this window or will we show success notification? Do we need redirect our user to new watchlist page?

Ask about it Andrey.
Show all cases in a small video.

**Comment by Partysun** [comment]
I see, you are working on it, add WIP label @devanshu-santiment

File: src/components/Watchlists/NewWatchlistForm.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import cx from 'classnames'
+import { Button, Toggle, Input } from '@santiment-network/ui'
+import * as actions from './../../actions/types'
+import styles from './NewWatchlistForm.module.scss'
+
+const MAX_NAME_LENGTH = 50
+const ERROR_MESSAGE =
```
Comment (haritonasty) [inline src/components/Watchlists/NewWatchlistForm.js]: max name length in mock-up: 25

File: src/components/Watchlists/NewWatchlistForm.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  'Something went wrong while creating watchlist, please try again!!!'
```
Comment (haritonasty) [inline src/components/Watchlists/NewWatchlistForm.js]: you show this error, when input is empty. In design "Create" button should be disabled for this case and without error

File: src/components/Watchlists/NewWatchlistForm.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            <Toggle
+              className={styles.toggle}
+              isActive={!isPublic}
+              onClick={this.togglePrivacy}
+            />{' '}
+            Secret
+          </div>
+          <div>
+            <Button border className={styles.cancel}>
+              Cancel
```
Comment (haritonasty) [inline src/components/Watchlists/NewWatchlistForm.js]: I think, this button should have onClick

File: src/components/Watchlists/NewWatchlistForm.js (hunk 2 above)
Comment (haritonasty) [reply src/components/Watchlists/NewWatchlistForm.js]: image
other colors here

**Comment by haritonasty** [comment]
in Figma other mock-up for mobile version

**Pull request closed without merging**